backend/reservations/views/rooms.py

```diff
@@ -240,6 +239,11 @@ def swap_gen(request):
             return Response("No guest found", status=status.HTTP_400_BAD_REQUEST)
 
         room = Room.objects.get(number=room_num, name_hotel='Ballys')
+
+        if not room.swappable():
+            return Response(f"Room {room.number} is not swappable",
+                            status=status.HTTP_400_BAD_REQUEST)
```

Comment:
I think by returning 400 here we go into the error case execution flow of the ux. which if i recall does not update the button state, just prints error msg to console. I could be wrong tho, would need look at the code and dont want to hold up this pr. i say merge it and risk is low enough that we send it. can try in staging if has time. ill look closer at the ux code later tonight if has chance

Reply:
Yes and there is _some_ logic in the frontend which controls what rooms folks see. Unlike the "my rooms" endpoint, I do not believe it toggles on individual rooms. Can see if we can land that in a subsequent PR later this week.